### PR TITLE
[NativeAOT-LLVM] Stub out finalization and enable some tests

### PR DESCRIPTION
**Pull request by SingleAccretion**
Finalization "support" available here is only that we'll run finalizers when explicitly requested - which is enough to enable the two tests I was looking to enable.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/nativeaot/Runtime/wasm/FinalizerHelpers.SingleThreaded.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+// This method is called at the end of GC in case finalizable objects were present.
+void RhEnableFinalization()
+{
+    // TODO: Implement automatic finalization. Cannot just call "ProcessFinalizersAndReturn"
+    // here as it will deadlock the GC.
+}
+
+EXTERN_C NATIVEAOT_API void __cdecl RhWaitForPendingFinalizers(UInt32_BOOL allowReentrantWait)
+{
+    // Must be called in cooperative mode as "ProcessFinalizersAndReturn" RPIs back into managed.
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/wasm/FinalizerHelpers.SingleThreaded.cpp]: Should this comment be other way around? The assert is verifying that it is not called in cooperative mode.

Comment (SingleAccretion) [reply src/coreclr/nativeaot/Runtime/wasm/FinalizerHelpers.SingleThreaded.cpp]: > Should this comment be other way around?

Yep. Fixed it.

**Review by yowl — APPROVED** [review]
Look good, thanks!

**Pull request merged**